### PR TITLE
Split Matter stack init apart from commissioner init in the Darwin framework

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
#### Problem
Right now we have a "init the controller" function that takes args that are only used the first time the function is called, which is very misleading.

#### Change overview
Make it clearer which things go with Matter stack init and which go with single-controller init.

#### Testing
The new semantics are unit tested by the XCTest bits.  Everything else should keep working and be covered by the normal tests.

### CODE REVIEW

File: src/darwin/Framework/CHIP/MatterStack.h [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+/*
+ * IPK to use for the controller's fabric.  Allowed to be null when
+ * starting a controller on an existing fabric.
+ */
+@property (strong, nonatomic, nullable) NSData * ipk;
+
+- (instancetype)init;
+@end
+
+@interface MatterStack : NSObject
```
Comment (anush-apple) [inline src/darwin/Framework/CHIP/MatterStack.h:95]:
```suggestion
@interface MatterControllerFactory : NSObject
```

File: src/darwin/Framework/CHIP/MatterStack.h [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+ * Fabric id for the controller.  Must be set to a nonzero value.
+ */
+@property (nonatomic) uint64_t fabricId;
+/*
+ * IPK to use for the controller's fabric.  Allowed to be null when
+ * starting a controller on an existing fabric.
+ */
+@property (strong, nonatomic, nullable) NSData * ipk;
+
+- (instancetype)init;
```
Comment (anush-apple) [inline src/darwin/Framework/CHIP/MatterStack.h:92]:
```suggestion

```

File: src/darwin/Framework/CHIP/MatterStack.h [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+@property (readonly, nonatomic) BOOL isRunning;
+
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+
+/**
+ * Return the single MatterStack we support existing.  It starts off
+ * in a "not started" state.
+ */
++ (MatterStack *)singletonStack;
```
Comment (anush-apple) [inline src/darwin/Framework/CHIP/MatterStack.h:106]:
```suggestion
+ (instancetype)sharedInstance;
```

File: src/darwin/Framework/CHIP/MatterStack.h [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+/*
+ * Whether to run a server capable of accepting incoming CASE
+ * connections.  Defaults to NO.
+ */
+@property (nonatomic) BOOL startServer;
+/*
+ * Path to a file to use for backing our KVS storage.  This should not
+ * be used, generally; it will be removed soon.  Defaults to nullptr;
+ */
+@property (nonatomic) const char * kvsPath;
```
Comment (anush-apple) [inline src/darwin/Framework/CHIP/MatterStack.h:63]:
```suggestion
@property (nonatomic, strong) NString * kvsPath;
```

File: src/darwin/Framework/CHIP/MatterStack.h [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+ * Create a CHIPDeviceController on a new fabric.  Returns nil on failure.
+ *
+ * This method will fail if the given fabric already exists.
+ *
+ * The fabric is identified by the root public key and fabric id in
+ * the startupParams.
+ */
+- (CHIPDeviceController * _Nullable)startControllerOnNewFabric:(CHIPDeviceControllerStartupParams *)startupParams;
+
+- (void)dealloc;
```
Comment (anush-apple) [inline src/darwin/Framework/CHIP/MatterStack.h:149]:
```suggestion
```

**Comment by bzbarsky-apple** [comment]
Fast-tracking platform specific change, but this needs review from one of the Apple folks.

**Review by anush-apple — APPROVED** [review]

**Pull request merged**